lib/src/typer.dart

```diff
@@ -48,14 +49,8 @@ class TyperAnimatedText extends AnimatedText {
   Widget animatedBuilder(BuildContext context, Widget child) {
     /// Output of CurveTween is in the range [0, 1] for majority of the curves.
     /// It is converted to [0, textCharacters.length].
-    var count = (_typingText.value * textCharacters.length).round();
-
-    if (count < 0) {
-      count = 0;
-    }
-    if (count >= textCharacters.length) {
-      count = textCharacters.length - 1;
-    }
+    final count =
+        (_typingText.value.clamp(0, 1) * textCharacters.length).round();
```

Comment:
Why are you rounding it? To get rid of negative value?

Reply:
`count` must be an integer. `final count = _typingText.value.clamp(0, 1) * textCharacters.length` actually returns a `double`. If I remove the `round()`, it immediately prompts the error `Another exception was thrown: type 'double' is not a subtype of type 'int'`.

Reply:
Ok this could be done by specifying datatype of `count`. But no issues with `round`.

Reply:
The `round()` is necessary because `count` is an _integer_ (not a _double_).